### PR TITLE
fix(HoverDisclosure): Remove strategy prop, add placeholderWidth

### PR DESCRIPTION
**Pull request by lukelooker**

### CODE REVIEW

File: packages/components/src/Tree/stories/FieldItem.tsx [hunk 1]
```diff
@@ -141,10 +142,7 @@ export const FieldItem: FC<FieldItemProps> = ({
             tooltipPlacement="top"
           />
         </HoverDisclosure>
-        <HoverDisclosure
-          visible={isFilter}
-          strategy={isPivot ? 'visibility' : 'display'}
-        >
+        <HoverDisclosure visible={isFilter} width={isPivot ? 24 : undefined}>
```
Comment (ElliotTheHuman) [inline packages/components/src/Tree/stories/FieldItem.tsx:145]: This looks good, but for some reason, the outermost flex container  above is shrinking this below 24px:
Screen Shot 2021-06-16 at 5 58 39 PM


However, looks like we can fix this by adding a div with `flex-basis: 24px` and `flex-shrink: 0`:
```
<div style={{ flexBasis: 24, flexShrink: 0 }}>
          <HoverDisclosure visible={isFilter} width={isPivot ? 24 : undefined}>
            <IconButton
              onClick={() => setIsFilter(!isFilter)}
              toggle={isFilter}
              toggleBackground
              toggleColor={color}
              icon={<FilterList />}
              label="Filter"
              tooltipPlacement="top"
            />
          </HoverDisclosure>
        </div>
```

**Review by ElliotTheHuman — APPROVED** [review]
Left a quick note about a `FieldItem` issue, but otherwise, I think we're good 👍

File: packages/components/src/Tree/stories/FieldItem.tsx (hunk 1 above)
Comment (lukelooker) [reply packages/components/src/Tree/stories/FieldItem.tsx:145]: Eww... Went ahead and built those props into the HoverDisclosure we'd need the `flexBasis` to move with the width prop as specified here.

Comment (lukelooker) [reply packages/components/src/Tree/stories/FieldItem.tsx:145]: Clarification... my "eww..." was in reference to the movement visible without the flexShrink in place

**Review by ElliotTheHuman — APPROVED** [review]
LGTM!

**Pull request merged**